Client/GeneralizedImageFetcher.swift

```diff
@@ -0,0 +1,49 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import UIKit
+import Shared
+import Logger
+
+struct GeneralizedImageFetcher {
+    /// This method is a generalized image fetcher that'll also cache. The completion is dispatched to `main`.
```

Comment:
I think we should make sure this is used only for avatar and other random images. Anything favicons or hero image should use `SiteImageView`. Maybe lets at least specify that in a comment?